src/nvim/fileio.c

```diff
@@ -6147,6 +6147,11 @@ void ex_doautoall(exarg_T *eap)
     /* restore the current window */
     aucmd_restbuf(&aco);
 
+    if (aco.new_curbuf->terminal) {
+      terminal_resize(aco.new_curbuf->terminal, aco.save_curwin->w_width,
+                      aco.save_curwin->w_height);
+    }
```

Comment:
This should probably go in `aucmd_restbuf`. The fact that the bug was only trigger in doautoall is just an accident isn't it?